example/src/main/scala/cilib/example/CSVExample.scala

```diff
@@ -0,0 +1,147 @@
+  val absoluteStream = Runner.staticProblem("absolute", absolute.eval)
+  val ackleyStream = Runner.staticProblem("ackley", ackley.eval)
+  val sphericalStream = Runner.staticProblem("spherical", spherical.eval)
+  val quadricStream = Runner.staticProblem("quadric", quadric.eval)
+
+  // Define the guides for our PSO algorithms
+  val cognitive = Guide.pbest[Mem[Double], Double]
+  val gbestGuide = Guide.gbest[Mem[Double]]
+  val lbestGuide = Guide.lbest[Mem[Double]](3)
+
+  // Define our algorithms
+  val gbestPSO = gbest(0.729844, 1.496180, 1.496180, cognitive, gbestGuide)
+  val lbestPSO = gbest(0.729844, 1.496180, 1.496180, cognitive, lbestGuide)
+
+  // Define iterators for the algorithms
+  val gbestIter = Iteration.sync(gbestPSO)
+  val lbestIter = Iteration.sync(lbestPSO)
+
+  // Define the initial swarm
+  val swarm =
+    Position.createCollection(PSO.createParticle(x => Entity(Mem(x, x.zeroed), x)))(bounds, 20)
+
+  // A class to hold the results that we want to save at the end of each iteration
+  final case class Results(min: Double, average: Double)
+
+  // A performance measure that we apply to the collection at the end of each iteration.
+  def performanceMeasure =
+    measure[NonEmptyList[Entity[Mem[Double], Double]], Unit, Results](collection => {
+
+      val fitnessValues = collection
+        .map(x =>
+          x.pos match {
+            case Point(_, _) => Double.PositiveInfinity
+            case Solution(_, _, o) =>
+              o match {
+                case Single(f, _) =>
+                  f match {
+                    case Feasible(v)      => v
+                    case Infeasible(_, _) => Double.PositiveInfinity
+                    case Adjusted(_, _)   => Double.PositiveInfinity
+                  }
+                case Multi(_) => Double.PositiveInfinity
+              }
+        })
+        .toList
+
+      Results(fitnessValues.min, fitnessValues.suml / fitnessValues.size)
+    })
+
+  // A simple RVar.pure function that is called when the the environment changes
+  // In this example our environments do not change.
+  val onChange = (x: NonEmptyList[Entity[Mem[Double], Double]], _: Eval[NonEmptyList, Double]) =>
+    RVar.pure(x)
+
+  def simulation(env: Environment[Double], stream: Process[Task, Problem[Double]]) =
+    List(
+      Runner.foldStep(env,
+                      rng,
+                      swarm,
+                      Runner.staticAlgorithm("GBestPSO", gbestIter),
+                      stream,
+                      onChange),
+      Runner.foldStep(env,
+                      rng,
+                      swarm,
+                      Runner.staticAlgorithm("LBestPSO", lbestIter),
+                      stream,
+                      onChange)
+    )
+
+  val simulations = List(
+    simulation(absolute, absoluteStream),
+    simulation(ackley, ackleyStream),
+    simulation(quadric, quadricStream),
+    simulation(spherical, sphericalStream)
+  ).flatten
```

Comment:
Although flatten does make sense, I'd prefer us to use `List.concat` here to make the intention that little bit clearer